Lift more info to Supported environments page

akeller opened the pull request:
## Description

Closes #3306. 

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->
<!-- Help the DevEx team prioritize our work (reviews, merges, etc.) by opening PRs sooner. -->

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [x] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

<!-- Keep in mind, Camunda maintains 18 months of versions. Backporting your change or including it in multiple versions is common. -->

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

akeller commented on docs/reference/supported-environments.md:
```diff
@@ -33,7 +33,25 @@ For example, 1.2+ means support for the minor version 2, and any higher minors (
+  - Number of nodes: `4`
+  - Volume type: `Performance (SSD) persistent disks`
+- [Red Hat OpenShift](/self-managed/platform-deployment/helm-kubernetes/platforms/redhat-openshift.md) - 4.11.x, 4.12.x, 4.13.x
+<!-- No current documentation available for recommended starting config.-->
```
Are there starting config recommendations for Red Hat? Currently, it's the only platform without them, but with specific supported versions.

theburi replied:
OpenShift does not have associated instance sizes as it could be installed on any infrastructure.

akeller replied:
I will adjust this so the expectations are clear and it doesn't look like a miss.

akeller commented on docs/reference/supported-environments.md:
```diff
@@ -50,35 +68,58 @@ Requirements for the components can be seen below:
+- [Amazon EKS](/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks#volume-performance)
+- [Google GKE](/self-managed/platform-deployment/helm-kubernetes/platforms/google-gke#volume-performance)
+- [Microsoft AKS](/self-managed/platform-deployment/helm-kubernetes/platforms/microsoft-aks#volume-performance)
+- [Red Hat OpenShift](#) <!-- No current documentation available. -->
```
OpenShift has no docs on volume performance. Do we need to add them?

akeller commented on docs/reference/supported-environments.md:
```diff
@@ -50,35 +68,58 @@ Requirements for the components can be seen below:
 
 Since the 8.4 release (January 2024), the Camunda Helm chart version is decoupled from the version of the application (e.g., the chart version is 9.0.0 and the application version is 8.4.x).
 
+<!-- Nit pick - this is a list, not a matrix. This is also on a separate site with no oversight from DevEx. -->
```
@MaxTru we should have a discussion about either collapsing these into docs.camunda.io or getting DevEx involved in reviews for helm.camunda.io.

MaxTru commented on docs/reference/supported-environments.md:
```diff
@@ -50,35 +68,58 @@ Requirements for the components can be seen below:
 You can also use newer versions of Desktop and Web Modeler with older Zeebe versions.
 :::
 
+## form-js version matrix
```
@akeller why is this needed from a customer perspective? Where is the source of this table?

akeller replied:
https://github.com/camunda/camunda-docs/pull/3300

This is the source. I merged it as a quick win, but we can move it elsewhere. I didn't have a better recommendation of where to put it.

MaxTru commented on docs/reference/supported-environments.md:
```diff
@@ -33,7 +33,25 @@ For example, 1.2+ means support for the minor version 2, and any higher minors (
-We highly recommend running Camunda 8 Self-Managed in a Kubernetes environment. We provide officially supported [Helm Charts](/self-managed/platform-deployment/helm-kubernetes/overview.md) for this. Please follow the [Installation Guide](/self-managed/platform-deployment/overview.md) to learn more about installation possibilities.
+We highly recommend running Camunda 8 Self-Managed in a Kubernetes environment. We provide officially supported [Helm Charts](/self-managed/platform-deployment/helm-kubernetes/overview.md) for this. Follow the [Installation Guide](/self-managed/platform-deployment/overview.md) to learn more about installation possibilities.
+
+The following list includes links for supported Kubernetes environments and recommended starting config. Generally speaking, the specification depends on your needs and workloads.
```
@akeller can you please make this more precise (or align with EMs to do so)? Is this the absolute minium requirement you always must have, or is the the "average", but you can be above or below based on workloads?

akeller replied:
AFAIK, this is average. Minimum would imply a smaller installation footprint (1 Zeebe vs. 3) and possible reduced availability.

MaxTru commented on docs/reference/supported-environments.md:
```diff
@@ -50,35 +68,58 @@ Requirements for the components can be seen below:
 
+### Volume Performance
+
+Depending on your cloud provider, we offer the following volume performance recommendations:
```
Also here, can we be specific please (recommendation vs minium requirement)? I would love the see minium requirements.

theburi replied:
We should define volume performance parameters to cover on-premises or "exotic cloud"  (i.e. Oracle, IBM, Digital Ocean, hosting providers)

theburi replied:
We come across questions about the support of NFS. In some environments, NFS could perform as well as local storage, and we need to define required performance characteristics and maybe call out some NFS's that we know are not supported.

akeller replied:
@theburi I view both of these items as feature enhancements and outside of the scope of this PR. Can you create epics to drive these changes with the appropriate team (Distribution or InfraEx)?

theburi replied:
Yes. Let's add this reference.

MaxTru commented on docs/reference/supported-environments.md:
```diff
@@ -33,7 +33,25 @@ For example, 1.2+ means support for the minor version 2, and any higher minors (
 ## Camunda 8 Self-Managed
 
-We highly recommend running Camunda 8 Self-Managed in a Kubernetes environment. We provide officially supported [Helm Charts](/self-managed/platform-deployment/helm-kubernetes/overview.md) for this. Please follow the [Installation Guide](/self-managed/platform-deployment/overview.md) to learn more about installation possibilities.
+We highly recommend running Camunda 8 Self-Managed in a Kubernetes environment. We provide officially supported [Helm Charts](/self-managed/platform-deployment/helm-kubernetes/overview.md) for this. Follow the [Installation Guide](/self-managed/platform-deployment/overview.md) to learn more about installation possibilities.
```
Should we explicitly link and state here which setups we support / not support? Basically what we have here: https://docs.camunda.io/docs/next/self-managed/platform-deployment/overview/#deployment-options

akeller replied:
I added links to the deployment options in my latest commit. We shouldn't document what we don't support. We should focus on what we do support.

MaxTru commented:
I added some comments. Overall: @akeller could you please try to rework this PR in a way that the user can easily read and understand what the minium requimrenets are to install, run, operate C8 SM? So focus on the "must-have" vs "recommendation"

akeller commented on docs/reference/supported-environments.md:
```diff
@@ -33,7 +33,25 @@ For example, 1.2+ means support for the minor version 2, and any higher minors (
+
+The following list includes links for supported Kubernetes environments and recommended starting config. Generally speaking, the specification depends on your needs and workloads.
+
+- [Amazon EKS](/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/amazon-eks.md)
```
Yes, we have some ways to manage redundancy with reusable comments. I added "
Identify content to move to reusable components" to the [issue tasklist](https://github.com/camunda/camunda-docs/issues/3306).

This page would be too big if we put all the information on a single page for all our supported environments.

theburi commented on docs/reference/supported-environments.md:
```diff
@@ -33,52 +33,108 @@ For example, 1.2+ means support for the minor version 2, and any higher minors (
+
+The following are the minimum supported requirements for cluster specification and dependencies:
+
+- 4vCPUs, 15 GB Memory
```
small note. vCPU will not be applicable on bare-metal installs. maybe we can call our "4 (v)CPU"

akeller replied:
Sure! @theburi Do we have documentation on bare-metal installs? Would this be [manual installation?](https://docs.camunda.io/docs/self-managed/platform-deployment/manual/)

theburi replied:
Generally, we would expect users to use Manual installation, but they could also use Docker-compose (Docker standalone). Looking at my comment today, CPU to vCPU relationship is hard to describe as there are different calculations. So I think it becomes too nuanced and we should not go there.

theburi commented on docs/reference/supported-environments.md:
```diff
@@ -33,52 +33,108 @@ For example, 1.2+ means support for the minor version 2, and any higher minors (
+- PostgreSQL 14.x or Amazon Aurora PostgreSQL 14.x
+- ...
+
+### Recommendations
```
Should we add a section for none-Kubernetes deployments? 
We still support Docker standalone (Compose) and Manual, which could be deployed on IaaS Cloud, VMs or Bare-metal.

akeller replied:
My understanding is that Docker Compose is not for production use, and I would not highlight it here. This should be production only. Please confirm you agree with the split [here](https://docs.camunda.io/docs/self-managed/platform-deployment/overview/#deployment-recommendation).

We've previously been very explicit about Helm as our recommended deployment path. Should we remove this and promote all deployment methods? This would be a change in direction in how we've presented this content.

theburi replied:
We officially support [4 production configurations](https://docs.camunda.io/docs/self-managed/platform-deployment/overview/#production) 
Let me clarify that we don't support docker-compose 'script' that deploys C8, but we support C8 deployed with Docker images (Which in reality would be done with customer-provided scripts using docker-compose similar to customer providing script to deploy JAR's)
We highly recommend HELM and Kubernetes, but our customers deploy with other options as well and we support them.

akeller commented:
After a working session with Max, I have more improvements to this page. I will cleanup the PR today, allow reviewers time on Monday and Tuesday, and then backport and merge Tuesday during US working hours. 

Related Slack thread - https://camunda.slack.com/archives/C026U8GBNSW/p1708101642758599

akeller commented:
@theburi FYI, during the working session, we removed some of the areas you made comments on in favor of deep links and did not expand the scope of the PR to include new platforms or deployment options. Please follow up on those with epics, as I mentioned.

MaxTru commented on docs/reference/supported-environments.md:
```diff
@@ -33,7 +39,35 @@ For example, 1.2+ means support for the minor version 2, and any higher minors (
+#### Volume Performance
+
+As a minimum requirement the cluster nodes should use volumes with an absolute minimum of 1.000 IOPS. **NFS volumes are not supported.** To ensure an appropriate sizing, ensure to [determine your influencing factors](../components/best-practices/architecture/sizing-your-environment.md#understanding-influencing-factors) (e.g., throughput), and conduct [benchmarking to validate an appropriate envir ...
+Refer to the following Cloud-provider specific examples, for typical volume types to be used.
+
+Depending on your cloud provider, we offer the following volume performance recommendations:
```
```suggestion
Refer to the following Cloud-provider specific examples, for typical volume types to be used:

```

MaxTru commented on docs/reference/supported-environments.md:
```diff
@@ -44,42 +78,49 @@ Requirements for the components can be seen below:
+### Component version matrix
 
-Since the 8.4 release (January 2024), the Camunda Helm chart version is decoupled from the version of the application (e.g., the chart version is 9.0.0 and the application version is 8.4.x).
+This overview shows which Zeebe version works with which Modeler, Operate, Tasklist, and Optimize:
```
```suggestion
This overview shows which Zeebe version works with which Modeler, Operate, Tasklist, and Optimize version:
```

akeller replied:
This sentence is still a little awkward. @christinaausley what would you do here?

hisImminence reviewed (dismissed):
/lgtm

akeller commented:
@MoAtCamunda will you have time to review this week? Can you review it as-is, or would it be easier if it was in our staging environment?

MaxTru commented on docs/reference/supported-environments.md:
```diff
@@ -4,6 +4,12 @@ title: "Supported environments"
 description: "Find out where to run Camunda 8 components for SaaS and Self-Managed, including Optimize for both Camunda 8 and Camunda 7."
 ---
 
+The supported environments page lists tested and compatible environments, deployment options, and component dependencies for successfully running Camunda 8.
```
This sentence reads strange to me. What about?

```suggestion
The supported environments page lists environments, deployment options, and component dependencies, which are tested and supported for compatibility with Camunda 8.
```

MoAtCamunda replied:
either way is fine for me

MaxTru commented on docs/reference/supported-environments.md:
```diff
@@ -4,6 +4,12 @@ title: "Supported environments"
 
+The supported environments page lists tested and compatible environments, deployment options, and component dependencies for successfully running Camunda 8.
+
+**If the particular technology is not listed, we are unable to resolve issues or provide guidance on integration.** You may [raise a feature request](/contact) that will be evaluated by our product teams to provide official support from Camunda.
```
Should we re-use the wording from bevore for clarity?
```suggestion
**If you are running Camunda 8 using an environment, deployment option, or component dependency that is not listed, we cannot resolve issues or provide guidance on integration.** You may [raise a feature request](/contact) that will be evaluated by our product teams to provide official support from Camunda.
```

MoAtCamunda replied:
Make sense from a consistency, but do we define environment, deployment option or component dependency anywhere? A customer will what to know what we mean. If we keep it as 'technology' we can use this a catch all which would be my preference instead of having to maintain more definitions

MaxTru replied:
The idea of this is to re-use the wording that is part of the page. The page would thereby give the definition. E.g., 

`Deployment Option`=> What a deployment option is, is defined as by the sub section of this page, which talks about `Deployment Options: HELM, Docker-compose, ...`.

@akeller can you please try to adjust this statement so that we pick up the terms of the sub-sections?

akeller replied:
"The supported environments page lists browsers, operating systems, clients, deployment options, and component requirements, which are tested and supported for compatibility with Camunda 8."

MaxTru commented on docs/reference/supported-environments.md:
```diff
@@ -4,6 +4,12 @@ title: "Supported environments"
+
+**If the particular technology is not listed, we are unable to resolve issues or provide guidance on integration.** You may [raise a feature request](/contact) that will be evaluated by our product teams to provide official support from Camunda.
+
+Recommendations are denoted with [recommended], however, other options are supported as well.
```
Maybe more verbose to ensure it is clearly understood what we mean:

```suggestion
If you started without any technology contraints, we advise you to follow our recommendations, which are denoted with [recommended]. However, the other listed options are supported as well.
```

akeller replied:
I like adding "the other listed options are supported as well," but I'm not sure we need the first half of the sentence to specify recommendations are for those without technology constraints. 

Since we use recommendations in other areas of our docs, we can add this as a glossary term and commit to what recommendations mean in the scope of Camunda docs.

MaxTru replied:
Okay. Can you please commit this proposal?

MaxTru commented on docs/reference/supported-environments.md:
```diff
@@ -33,7 +39,36 @@ For example, 1.2+ means support for the minor version 2, and any higher minors (
+- [Docker](/self-managed/platform-deployment/docker.md)
+- [Manual](/self-managed/platform-deployment/manual.md)
+
+Ensure the Camunda component versions are compatible with the Helm chart version as defined in the [matrix](https://helm.camunda.io/camunda-platform/version-matrix/).
```
This statement is only valid for the "HELM" option, yet it reads and is placed as if it would apply to the entire section. Can you please check whether we cannot put it under the - HELM list point?

akeller replied:
I moved this to an admonition titled "Helm chart compatibility".

Adding it to the list would make for a disjointed list.

MoAtCamunda commented on versioned_docs/version-8.4/reference/supported-environments.md:
```diff
@@ -33,7 +39,36 @@ For example, 1.2+ means support for the minor version 2, and any higher minors (
 
 ## Camunda 8 Self-Managed
 
-We highly recommend running Camunda 8 Self-Managed in a Kubernetes environment. We provide officially supported [Helm Charts](/self-managed/platform-deployment/helm-kubernetes/overview.md) for this. Please follow the [Installation Guide](/self-managed/platform-deployment/overview.md) to learn more about installation possibilities.
```
We recommend running Camunda 8 Self-Managed in a Kubernetes environment. We provide officially supported [Helm Charts](/self-managed/platform-deployment/helm-kubernetes/overview.md) for this. Please follow the [Installation Guide](/self-managed/platform-deployment/overview.md) to learn more about installation possibilities.

MoAtCamunda replied:
Suggest we remove highly as its ambiguous (either we make this mandatory or optional but not highly

akeller replied:
This is language pulled from the existing content in the Self-Managed section. I will follow up in a separate PR to change that as well.

theburi commented on versioned_docs/version-8.3/reference/supported-environments.md:
```diff
@@ -33,49 +39,86 @@ For example, 1.2+ means support for the minor version 2, and any higher minors (
-We highly recommend running Camunda 8 Self-Managed in a Kubernetes environment. We provide officially supported [Helm Charts](/self-managed/platform-deployment/helm-kubernetes/overview.md) for this. Please follow the [Installation Guide](/self-managed/platform-deployment/overview.md) to learn more about installation possibilities.
+### Deployment options
+
+- [Helm/Kubernetes](/self-managed/platform-deployment/helm-kubernetes/overview.md) [recommended]
```
Link to K8S certification link [https://www.cncf.io/training/certification/software-conformance/](https://www.google.com/url?q=https://www.cncf.io/training/certification/software-conformance/&sa=D&source=docs&ust=1708512503126547&usg=AOvVaw2RLHpL0yFSylGOH-mfdFYj)

akeller replied:
I linked to the subsection on benefits since I thought that was most relevant for our customers who weren't familiar with Certified Kubernetes - https://www.cncf.io/training/certification/software-conformance/#benefits

akeller commented:
@MaxTru @MoAtCamunda @theburi, I've addressed feedback from the PR review, sync discussions, and the document in the last several commits. I've also backported the changes to 8.3 and 8.4. 

I reworked the first sentence for @MaxTru and @MoAtCamunda to include all H2 sections below (minus the last two matrices on the page.

I added the Helm chart matrix into an admonition for @MaxTru since it doesn't fit to include the info in the list (the list is a set of deployment options). 

You'll see changes to the Deployment options section to include requests from @theburi:
* Add info on Certified Kubernetes
* Add info from [Kubernetes environments](https://docs.camunda.io/docs/self-managed/platform-deployment/helm-kubernetes/overview/#kubernetes-environments), including Stock Kubernetes.
  * I did not add KIND (or similar) as it's only for local development

I did not take action on the sentence defining `[recommended]`, see https://github.com/camunda/camunda-docs/pull/3310#discussion_r1497043946. 
  
I will follow up on any redundant content in a future PR. Let's get this page settled first and I'll work backward to update the other pages across the Self-Managed section. 

One point of clarification - do we want to remove "highly recommend" as language we use in the docs entirely? Are there reserve words or words we'd like to exclude from the Supported Environments page? Your thoughts would be helpful here and for guidance on future PR reviews.

MaxTru commented:
> One point of clarification - do we want to remove "highly recommend" as language we use in the docs entirely? Are there reserve words or words we'd like to exclude from the Supported Environments page? Your thoughts would be helpful here and for guidance on future PR reviews.

@akeller my opinion: yes, let's remove the word "highly" entirely. Either it is recommended (recommended defined as stated in my other [comment](https://github.com/camunda/camunda-docs/pull/3310#discussion_r1497043946)) or not. It is unclear what the addition "highly" would mean.

MoAtCamunda commented:
> One point of clarification - do we want to remove "highly recommend" as language we use in the docs entirely? Are there reserve words or words we'd like to exclude from the Supported Environments page? Your thoughts would be helpful here and for guidance on future PR reviews.

I would suggest that we remove "highly recommend" for all docs. I dont have other suggestions for other keywords

akeller commented:
Adding the `hold` label since more comments came in on the Google doc and may impact changes here.

I'll plan to cleanup the merge conflicts later today or Monday.

theburi commented on docs/reference/supported-environments.md:
```diff
@@ -33,7 +39,42 @@ For example, 1.2+ means support for the minor version 2, and any higher minors (
+
+The following are tested and supported deployment options for Kubernetes, Docker, and manual installation:
+
+- [Stock Kubernetes](/self-managed/platform-deployment/helm-kubernetes/overview.md)
```
We need to clarify what does it mean -Stock Kubernetes. It is not a standard term that used in CNCF. Does it mean upstream or certified?
imo we should support any certified distribution and specifically highlight tested platforms. (In a similar fashion to Elastic or Confluence)

christinaausley replied:
This is a great callout. I think this should likely just be referenced as Kubernetes (k8s).

We can add style guide assistance on this once agreed upon here.

theburi replied:
ok. Let's keep it as is, so we can progress here, and I will clarify and make another Epic to clarify this specific point and we will make changes throughout the docs.

megglos commented on docs/reference/supported-environments.md:
```diff
@@ -33,7 +39,42 @@ For example, 1.2+ means support for the minor version 2, and any higher minors (
+
+#### Volume performance
+
+As a minimum requirement the cluster nodes should use volumes with an absolute minimum of 1.000 IOPS. **NFS volumes are not supported.**
```
It recently came up whether we support other network storage types, I would expand on stating no network storage is supported.

```suggestion
As a minimum requirement the cluster nodes should use volumes with an absolute minimum of 1.000 IOPS. **NFS or other types of network storage volumes are not supported.**
```

akeller replied:
@megglos I will also follow up in a separate PR to update this on other pages.

megglos commented on versioned_docs/version-8.3/reference/supported-environments.md:
```diff
@@ -33,49 +39,92 @@ For example, 1.2+ means support for the minor version 2, and any higher minors (
+#### Volume performance
 
-The core Camunda components have a unified fixed release schedule following the [release policy](./release-policy.md). However, some of the applications have their own schedule. The following compatibility matrix gives an overview of the different versions with respect to the Helm chart versions.
+As a minimum requirement the cluster nodes should use volumes with an absolute minimum of 1.000 IOPS. **NFS volumes are not supported.**
```
```suggestion
As a minimum requirement the cluster nodes should use volumes with an absolute minimum of 1.000 IOPS. **NFS or other types of network storage volumes are not supported.**
```

megglos commented on versioned_docs/version-8.4/reference/supported-environments.md:
```diff
@@ -33,7 +39,42 @@ For example, 1.2+ means support for the minor version 2, and any higher minors (
+
+#### Volume performance
+
+As a minimum requirement the cluster nodes should use volumes with an absolute minimum of 1.000 IOPS. **NFS volumes are not supported.**
```
```suggestion
As a minimum requirement the cluster nodes should use volumes with an absolute minimum of 1.000 IOPS. **NFS or other types of network storage volumes are not supported.**
```

megglos reviewed (approved):
LGTM! One suggestion for stating no types of network based storage is supported.

MaxTru commented on docs/reference/supported-environments.md:
```diff
@@ -4,7 +4,13 @@ title: "Supported environments"
-:::note
+The supported environments page lists browsers, operating systems, clients, deployment options, and component requirements, which are tested and supported for compatibility with Camunda 8.
+
+**If the particular technology is not listed, we are unable to resolve issues or provide guidance on integration.** You may [raise a feature request](/contact) that will be evaluated by our product teams to provide official support from Camunda.
```
Align with latest discussions in https://docs.google.com/document/d/1PjxMqicBKNCll-7EyJCga7n3NkWWMeyXgwfdyZ3D7eU/edit

Proposal:
```suggestion
**If the particular technology is not listed, it may cause issues for Camunda, which we cannot resolve.** If you rely on a technology that is not supported, you may [raise a feature request](/contact) that will be evaluated by our product teams to provide official support from Camunda, or [raise a help request](/contact) to request Consulting services.
```

akeller replied:
Let's just keep this brief and more open-ended - if the particular technology is not listed, and you're looking for support OR guidance (like Best Practices), we wouldn't be able to help outside of a consulting request. 

```suggestion
**If the particular technology is not listed, we cannot resolve issues or guide integrations.** 

You may [raise a feature request](/contact) that will be evaluated by our product teams to provide official support from Camunda, or you can make a [help request](/contact) to work with Consulting services. 
```

I'm hesitant to add "guide integrations" here because we offer possible guidance through Consulting, but it's up to their discretion. I'd like to see this clearly defined in our contracts because the way it currently reads, it sounds like Consulting will support anything and everything via help requests, and that may not be true.

akeller replied:
```suggestion
**If the particular technology is not listed, we cannot resolve issues caused by the usage of that unlisted technology.**

You may [raise a feature request](/contact) that will be evaluated by our product teams to provide official support from Camunda, or you can make a [help request](/contact) to work with Consulting services. 
```

MaxTru replied:
Agreed. Let's do that.

akeller commented:
@christinaausley can I get a 👍 from you so we can marge this PR? 🎉
